### ISSUE TITLE
Reg. F.normalize dim=0 in feat = F.normalize(self.head(feat), dim=0)

### ISSUE THREAD

**Issue by phaniparsa** [CaixdLab/CL4CaPro#2]
Hi All,

On line#29 in https://github.com/CaixdLab/CL4CaPro/blob/main/CLCPnet.py file, for L2 normalization dim value is given as 0.

May I please know if there any specific reason for doing feature-wise L2-norm instead of sample-wise L2-norm?

Thank you!

**Comment by anchensun** [comment]
For an extreme high‑dimensional, low‑sample gene‑expression scenario trained in a single full batch, we believe dim=0 L2‑normalisation is a defensible design choice: it equalises every gene’s influence, substitutes for batch‑level standardisation, aligns with sparsity‑oriented genomics objectives, and matches the assumptions of redundancy‑reduction contrastive losses.
